src/app/api/league.py

```diff
@@ -33,15 +37,28 @@ class LeagueAPI(BaseAuthResource):
     def get(self, league_id):
         """ return the specified league """
         league = LeagueModel.build_key(league_id=league_id, user_key=self.user.key).get()
-        return {'data': marshal(league, league_template)}
+        if not league:
+            return 'Invalid league.', 404
+        return marshal(league, league_template)
 
     def put(self, league_id):
         """ update the specified league """
-        updated_league = update_league(self.user, league_id, self.args.get('name'), self.args.get('rating_scheme'),
-                                       description=self.args.get('description'))
-        return {'data': marshal(updated_league, league_template)}
+        try:
+            updated_league = update_league(self.user, league_id, self.args.get('name'), self.args.get('rating_scheme'),
+                                           description=self.args.get('description'))
+        except ValueError:
```

Comment:
you shouldn't catch the ValueError. Under normal circumstances that's never going to be raised right? the base class will ensure league_id is non-null